### PR TITLE
Upgrade dependencies ahead of v1.34.0

### PR DESCRIPTION
**Pull request by AndrewSirenko**
**Is this a bug fix or adding new feature?**

/cleanup

**What is this PR about? / Why do we need it?**

Upgrade dependencies ahead of v1.34.0

**What testing is done?** 

make test; make verify

**Special Notes**

Only upgraded to k8s v0.30.3 dependencies due to too many issues in v1.31.0. 

Had to change that error formatting in node.go due to linting.

### CODE REVIEW

**Review by torredil — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -14,45 +14,45 @@ require (
...
 	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.28.0
 	go.opentelemetry.io/otel/sdk v1.28.0
-	golang.org/x/sys v0.22.0
+	golang.org/x/sys v0.24.0
 	google.golang.org/grpc v1.65.0
 	google.golang.org/protobuf v1.34.2
-	k8s.io/api v0.31.0-beta.0
-	k8s.io/apimachinery v0.31.0-beta.0
+	k8s.io/api v0.31.0-alpha.2
+	k8s.io/apimachinery v0.31.0-alpha.2
```
Comment (ConnorJC3) [inline go.mod:30]: Please switch these versions to `v0.31.0` (no alpha)

File: go.mod [hunk 2]
```diff
@@ -152,36 +151,36 @@ require (
...
+	k8s.io/kubectl => k8s.io/kubectl v0.30.3
+	k8s.io/kubelet => k8s.io/kubelet v0.30.3
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.30.3
+	k8s.io/metrics => k8s.io/metrics v0.30.3
+	k8s.io/mount-utils => k8s.io/mount-utils v0.30.3
+	k8s.io/node-api => k8s.io/node-api v0.30.3
+	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.30.3
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.30.3
+	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.30.3
+	k8s.io/sample-controller => k8s.io/sample-controller v0.30.3
```
Comment (ConnorJC3) [inline go.mod:183]: These should all be bumped to `v0.31.0` to match the other k8s dependencies we have.

File: hack/tools/install.sh [hunk 3]
```diff
@@ -19,27 +19,27 @@ set -euo pipefail
...
-GOLANGCI_LINT_VERSION="v1.59.1"
+GOLANGCI_LINT_VERSION="v1.60.1"
 # https://github.com/hairyhenderson/gomplate
 GOMPLATE_VERSION="v4.1.0"
 # https://github.com/helm/helm
-HELM_VERSION="v3.15.3"
+HELM_VERSION="v3.15.4"
 # https://github.com/kubernetes/kops
-KOPS_VERSION="v1.29.0"
+KOPS_VERSION="v1.30.0"
```
Comment (ConnorJC3) [inline hack/tools/install.sh:36]: Backout `kops` upgrade due to `kops` bugs

**Review by ConnorJC3 — CHANGES_REQUESTED** [review]

**Pull request merged**